api/controllers/resource_controller.js

```diff
@@ -1,141 +1,240 @@
+        errors.push(err)
     }
+    return {
+        updated_resources: updated_resources,
+        errors: errors,
+    };
 }
 
-exports.get_resources_by_customer_id = (req, res) => {
+exports.get_resources_by_customer_id = async (req, res) => {
     let message = "";
     let status = 200;
     let id = req.user_id;
+    let errors = [];
 
-    // Updates status of a users resources known to spark
-    update_resources();
-
-    Resources.find({owner: id}, (err, resources) => {
-        if(err) {
-            message = `There was an error while retrieving your resources.\nError: ${err.name}`;
-            status = 500;
-        } else {
-            message = "Resources retrieved successfully.";
-        }
-        res.status(status).json({
-            success: !!resources,
-            error: err ? err : null,
+    let resources = null;
+    try {
+        // Updates status of a users resources known to spark
+        // let {errors: errs} = await update_resources_from_spark();
+
+        // if (errs.length !== 0) {
+        //     errors.push(new Error("Unable to update resources from cluster"));
+        // }
+
+        resources = await Resources.find({owner: id});
+
+        message = "Successfully retrieved your resources.";
+
+    } catch (err) {
+        message = `There was an error while retrieving your resources.`;
+        status = 500;
+        errors.push(err);
+    } finally {
+        return res.status(status).json({
+            // If we have a new resource and there were no errors this was successful
+            success: !!resources && (errors.length === 0),
+            errors: errors,
             message: message,
-            resources: resources,
+            data: resources,
         });
-    });
-
+    }
 };
 
-exports.add_resource_by_customer_id = (req, res) => {
+exports.add_resource_by_customer_id = async (req, res) => {
     let message, resource;
     let status = 200;
     let id = req.user_id;
+    let errors = [];
+    let new_resource = null;
 
-    // For some reason this has to be initialized earlier to work.
-    resource = new Resources({
-        ip_address: req.body.ip_address,
-        ram: req.body.ram,
-        cores: req.body.cores,
-        cpus: req.body.cpus,
-        gpus: req.body.gpus,
-        status: req.body.status,
-        price: req.body.price,  // TODO: this may need to be determined server side
-        owner: id,
-        createdBy: id,
-        updatedBy: id,
-        machine_name: req.body.machine_name
-    });
+    try {
+        // TODO: Move new resource creation to its own try/catch block
+        // For some reason this has to be initialized earlier to work.
+        resource = await new Resources({
+            ip_address: req.body.ip_address,
+            ram: req.body.ram,
+            cores: req.body.cores,
+            cpus: req.body.cpus,
+            gpus: req.body.gpus,
+            status: req.body.status,
+            price: req.body.price,
+            owner: id,
```

Comment:
here can be `req.user_id` (just format the code on the same way)

Reply:
I actually want to use `id` here as we can unpack its value into a `const` and (hopefully) have some safety against accidentally modifying it.